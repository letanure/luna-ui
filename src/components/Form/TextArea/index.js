import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const TextArea = styled.textarea`
  width: 100%;
  border: none;
  outline: none;
  padding: 20px;
  font-size: 14px;
  box-shadow: none;
  height: ${(props) => props.height};
  transition: all .25s linear;
  ${(props) => !props.resize ? 'resize: none;' : ''}
  background-color: ${(props) => props.theme.form.textarea.background};
  border: 1px solid ${(props) => props.theme.form.textarea.border};
  border-radius: 2px;
  ::-webkit-input-placeholder { /* Chrome/Opera/Safari */
    color: ${(props) => props.theme.form.textarea.placeholder};
  }
  ::-moz-placeholder { /* Firefox 19+ */
    color: ${(props) => props.theme.form.textarea.placeholder};
  }
  :-ms-input-placeholder { /* IE 10+ */
    color: ${(props) => props.theme.form.textarea.placeholder};
  }
  :-moz-placeholder { /* Firefox 18- */
    color: ${(props) => props.theme.form.textarea.placeholder};
  }
  &:focus {
    border-color: ${(props) => props.theme.form.textarea.types[props.kind]};
  }
  &:disabled {
    cursor: not-allowed;
    color: ${(props) => props.theme.form.textarea.disabledText};
    background-color: ${(props) => props.theme.form.textarea.disabled};
  }
`;

/**
 * TextArea component.
 * E.g.:
 * ```html
 * <Form.TextArea kind="default" height='80px' resize />
 * ```
 */
class TextAreaComponent extends React.PureComponent {
  render() {
    const { kind, config, height, resize, children, className } = this.props;
    return (
      <TextArea className={className} kind={kind} height={height} resize={resize} {...config}>{children}</TextArea>
    );
  }
}

TextAreaComponent.propTypes = {
  children: PropTypes.node,
  kind: PropTypes.string,
  config: PropTypes.object,
  height: PropTypes.string,
  resize: PropTypes.bool,
  className: PropTypes.string
};

TextAreaComponent.defaultProps = {
  children: null,
  kind: 'primary',
  config: {},
  height: 200,
  resize: false,
  className: ''
};

export default TextAreaComponent;
