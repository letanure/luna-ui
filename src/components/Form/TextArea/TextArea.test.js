import React from 'react';
import sinon from 'sinon';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import theme from '../../../config/defaultTheme';
import ThemeProvider from '../../Theme/index';

import TextArea from './index';

describe('<TextArea />', () => {
  let textAreaList;
  let onClick;

  beforeEach(() => {
    onClick = sinon.spy();

    textAreaList =
      <ThemeProvider theme={theme}>
        <TextArea kind="default" height="80px" resize config={{ defaultValue: 'Bla is blu when blu is bla' }} />
        <TextArea kind="default" height="80px" resize config={{ onClick }} />
      </ThemeProvider>;
  });

  test('Check options deafult value', () => {
    const wrapper = mount(textAreaList);
    expect(wrapper.find('textarea').at(0)
      .getDOMNode().value).toBe('Bla is blu when blu is bla');
  });

  test('Call onClick', () => {
    const wrapper = mount(textAreaList);
    const textarea = () => wrapper.find('textarea').at(1);
    textarea().simulate('click');
    expect(onClick).toHaveProperty('callCount', 1);
  });

  test('toMatchSnapshot', () => {
    const tree = renderer
      .create(textAreaList)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
