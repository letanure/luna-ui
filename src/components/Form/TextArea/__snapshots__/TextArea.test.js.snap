// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArea /> toMatchSnapshot 1`] = `
<div>
  <textarea
    className="sc-bdVaJa eebKDZ"
    defaultValue="Bla is blu when blu is bla"
    height="80px"
    kind="default"
  />
  <textarea
    className="sc-bdVaJa eebKDZ"
    height="80px"
    kind="default"
    onClick={[Function]}
  />
</div>
`;
